src/Storages/MeiliSearch/StorageMeiliSearch.cpp

```diff
@@ -0,0 +1,161 @@
+#include <Parsers/ASTSelectQuery.h>
+#include <QueryPipeline/Pipe.h>
+#include <Storages/IStorage.h>
+#include <Storages/MeiliSearch/MeiliSearchConnection.h>
+#include <Storages/MeiliSearch/SinkMeiliSearch.h>
+#include <Storages/MeiliSearch/SourceMeiliSearch.h>
+#include <Storages/MeiliSearch/StorageMeiliSearch.h>
+#include <Storages/SelectQueryInfo.h>
+#include <Storages/StorageFactory.h>
+#include <Storages/StorageInMemoryMetadata.h>
+#include <Storages/transformQueryForExternalDatabase.h>
+#include <base/logger_useful.h>
+#include <Common/parseAddress.h>
+#include "Core/Types.h"
+#include "Parsers/ASTFunction.h"
+#include "Parsers/IAST_fwd.h"
+#include "Processors/Formats/IOutputFormat.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int BAD_QUERY_PARAMETER;
+}
+
+StorageMeiliSearch::StorageMeiliSearch(
+    const StorageID & table_id,
+    const MeiliSearchConfiguration & config_,
+    const ColumnsDescription & columns_,
+    const ConstraintsDescription & constraints_,
+    const String & comment)
+    : IStorage(table_id), config{config_}, log(&Poco::Logger::get("StorageMeiliSearch (" + table_id.table_name + ")"))
+{
+    StorageInMemoryMetadata storage_metadata;
+    storage_metadata.setColumns(columns_);
+    storage_metadata.setConstraints(constraints_);
+    storage_metadata.setComment(comment);
+    setInMemoryMetadata(storage_metadata);
+}
+
+void printAST(ASTPtr ptr)
+{
+    WriteBufferFromOwnString out;
+    IAST::FormatSettings settings(out, true);
+    settings.identifier_quoting_style = IdentifierQuotingStyle::BackticksMySQL;
+    settings.always_quote_identifiers = IdentifierQuotingStyle::BackticksMySQL != IdentifierQuotingStyle::None;
+    ptr->format(settings);
+}
+
+std::string convertASTtoStr(ASTPtr ptr)
```

Comment:
Also as this PR is very close to be merged now -- please remove these debug functions `printAST` and `convertASTToStr`.

Reply:
OK, however I can't remove convertASTToStr as this is commonly used for logging.